runtime/transformations/components/visualization/time_series_and_value_intervals.py

```diff
@@ -0,0 +1,329 @@
+    - If any value interval in `value_interval_dict` contains invalid inputs.
+
+## Examples
+
+Example json input for a call of this component is:
+```
+{
+    "series": {
+        "2020-01-01T01:15:27.000Z": 42.0,
+        "2020-01-01T16:20:00.000Z": 16.1,
+        "2020-01-02T08:20:00.000Z": 27.0,
+        "2020-01-02T16:20:00.000Z": 2.6,
+        "2020-01-03T08:20:04.000Z": 25.9
+    },
+    "value_interval_dict": {
+        "[10,20]": {
+            "min_value": 10.0,
+            "max_value": 20.0,
+            "min_value_inclusive": true,
+            "max_value_inclusive": true,
+            "display_name": "Intv. 1"
+        },
+        "(20,30)": {
+            "min_value": 20.0,
+            "max_value": 30.0,
+            "min_value_inclusive": false,
+            "max_value_inclusive": false,
+            "line_color": "rgb(40,20,250)",
+            "fill_color": "rgba(40,20,250,0.15)"
+        }
+    },
+    "line_color": "rgb(40, 20, 250)",
+    "fill_color": "rgba(40, 20, 250, 0.15)"
+}
+```
+
+"""
+
+
+import logging
+
+import pandas as pd
+import plotly.graph_objects as go
+import plotly.io as pio
+from pydantic import BaseModel, ValidationError, root_validator
+
+from hetdesrun.runtime.exceptions import ComponentInputValidationException
+from hetdesrun.utils import plotly_fig_to_json_dict
+
+pio.templates.default = None
+
+logger = logging.getLogger(__name__)
+
+
+class ValueInterval(BaseModel):
+    min_value: float
+    min_value_inclusive: bool = True
+    max_value: float
+    max_value_inclusive: bool = True
+    fill_color: str | None
+    line_color: str | None
+    display_name: str | None
+
+    @root_validator()
+    def verify_value_ranges(cls, values: dict) -> dict:
+        min_value = values["min_value"]
+        min_value_inclusive = values["min_value_inclusive"]
+        max_value = values["max_value"]
+        max_value_inclusive = values["max_value_inclusive"]
+        if max_value < min_value:
+            raise ValueError(
+                "To be valid, a value interval must be non-empty, i.e. min_value may not "
+                "be bigger than max_value."
+            )
+
+        if (min_value_inclusive is False or max_value_inclusive is False) and not (
+            min_value < max_value
+        ):
+            raise ValueError(
+                "To be valid, a value interval must be non-empty, i.e min_value must "
+                "be smaller than max_value."
+            )
+        return values
+
+
+def plot_series_and_ranges(
+    series: pd.Series,
+    interval_dict: dict[str, ValueInterval],
+    line_color: str,
+    fill_color: str,
+) -> go.Figure:
+    series_to_plot = series.sort_index()
+    fig = go.Figure(
+        [go.Scatter(x=series_to_plot.index, y=series_to_plot, mode="lines+markers")]
+    )
+
+    time_delta = series_to_plot.index.max() - series_to_plot.index.min()
+    num_intervals = len(interval_dict)
+
+    for interval_key in interval_dict:
```

Comment:
Iterate over interval_dict.items() to directly get access to the values as well:
`for interval_key, interval_value in interval_dict.items():`
Consequently, you should replace `interval_dict[interval_key]` with `interval_value` within the for loop.